src/ir/operation.cc

```diff
@@ -23,22 +23,43 @@
 
 namespace raksha::ir {
 
+// Note: I tried to make this only defined in terms of the type parameter T, but
+// the compiler was unhappy with me writing std::function<std::string(const T&)>
+// and I could not figure out how to placate it.
+template<class T, class U>
+static std::string PrintNamedMapInNameOrder(
+    const absl::flat_hash_map<std::string, T> &map_to_print,
+    U value_printing_fn) {
+  std::vector<absl::string_view> names;
+  names.reserve(map_to_print.size());
+  for (auto &key_value_pair : map_to_print) {
+    names.push_back(key_value_pair.first);
+  }
+  std::sort(names.begin(), names.end());
```

Comment:
Insert into an  absl::btree_set to get sorted right away?

Reply:
Usually, I like to do offline sorts when possible, as it tends to perform better. But in this circumstance, where it is plausible that the value is printed out multiple times and it is much more concise, I think perhaps your suggestion of a `btree_set` makes more sense.

Reply:
Sounds good.